### PR TITLE
Clowder jenkinsfile PR Check Enhancement

### PR DESCRIPTION
**Pull request by maknop**

### CODE REVIEW

**Review by Victoremepunto — CHANGES_REQUESTED** [review]
I'm sorry, I don't think this is in an appropriate state to be merged.

There are several spots where I don't quite understand the intention behind the changes or if the changes are actually an improvement compared to the original script.

I left comments to ask for clarification in all these places, please feel free to schedule some time to review this work together if needed.

File: ci/unit_tests.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+curl -s ${HELPER_FUNCTIONS} > .general.sh && source "./.general.sh"
```
Comment (Victoremepunto) [inline ci/unit_tests.sh]: is this still being used ? if it's not please let's remove it.

File: ci/minikube_e2e_tests.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
+
+IMG=$IMAGE_NAME:$IMAGE_TAG BASE_IMG=$BASE_IMG make docker-build
+IMG=$IMAGE_NAME:$IMAGE_TAG make docker-push
+
+tempid=$(docker create "$IMAGE_NAME:$IMAGE_TAG")
+docker cp "${tempid}:/manifest.yaml" .
+docker rm $tempid
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:18]: quotes 
```suggestion
docker rm "$tempid"
```

File: ci/minikube_e2e_tests.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+set -x
+
+set -exv
+
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
+
+IMG=$IMAGE_NAME:$IMAGE_TAG BASE_IMG=$BASE_IMG make docker-build
+IMG=$IMAGE_NAME:$IMAGE_TAG make docker-push
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:14]:
```suggestion
IMG="${IMAGE_NAME}:${IMAGE_TAG}" BASE_IMG="$BASE_IMG" make docker-build
IMG="${IMAGE_NAME}:${IMAGE_TAG}" make docker-push
```

File: ci/build_push_base_img.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+echo "received HTTP response: $RESPONSE"
+
+# find all non-expired tags
+VALID_TAGS_LENGTH=$(echo $RESPONSE | jq '[ .tags[] | select(.end_ts == null) ] | length')
+
+# Check if Clowder's base image tag already exists
+if [[ "$VALID_TAGS_LENGTH" -eq 0 ]]; then
+    BASE_IMG=$BASE_IMG make docker-build-and-push-base
```
Comment (Victoremepunto) [inline ci/build_push_base_img.sh:21]:
```suggestion
    BASE_IMG="$BASE_IMG" make docker-build-and-push-base
```

File: ci/minikube_e2e_tests.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+set -x
+
+set -exv
+
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
+
+IMG=$IMAGE_NAME:$IMAGE_TAG BASE_IMG=$BASE_IMG make docker-build
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:13]:
```suggestion
IMG="${IMAGE_NAME}:${IMAGE_TAG}" BASE_IMG="$BASE_IMG" make docker-build
```

File: ci/minikube_e2e_tests.sh (hunk 3 above)
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:14]:
```suggestion
IMG="${IMAGE_NAME}:${IMAGE_TAG}" make docker-push
```

File: ci/minikube_e2e_tests.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+IMG=$IMAGE_NAME:$IMAGE_TAG make docker-push
+
+tempid=$(docker create "$IMAGE_NAME:$IMAGE_TAG")
+docker cp "${tempid}:/manifest.yaml" .
+docker rm $tempid
+
+docker build -f Dockerfile.test --build-arg BASE_IMAGE=${BASE_IMG} -t $CONTAINER_NAME .
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:20]:
```suggestion
docker build -f Dockerfile.test --build-arg BASE_IMAGE="$BASE_IMG" -t "$CONTAINER_NAME" .
```

File: ci/minikube_e2e_tests.sh [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    -e IMAGE_NAME=$IMAGE_NAME \
+    -e IMAGE_TAG=$IMAGE_TAG \
+    -e QUAY_USER=$QUAY_USER \
+    -e QUAY_TOKEN=$QUAY_TOKEN \
+    -e MINIKUBE_HOST=$MINIKUBE_HOST \
+    -e MINIKUBE_ROOTDIR=$MINIKUBE_ROOTDIR \
+    -e MINIKUBE_USER=$MINIKUBE_USER \
+    -e CLOWDER_VERSION=$CLOWDER_VERSION \
+    $BASE_IMG \
+    ./ci/minikube_e2e_tests_inner.sh
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:35]:
```suggestion
docker run -i \
    --name "$CONTAINER_NAME" \
    -v "${PWD}:/workspace:ro" \
    -v "$($PWD/bin/setup-envtest use -p path):/bins:ro" \
    -e IMAGE_NAME="$IMAGE_NAME" \
    -e IMAGE_TAG="$IMAGE_TAG" \
    -e QUAY_USER="$QUAY_USER" \
    -e QUAY_TOKEN="$QUAY_TOKEN" \
    -e MINIKUBE_HOST="$MINIKUBE_HOST" \
    -e MINIKUBE_ROOTDIR="$MINIKUBE_ROOTDIR" \
    -e MINIKUBE_USER="$MINIKUBE_USER" \
    -e CLOWDER_VERSION="$CLOWDER_VERSION" \
    "$BASE_IMG" \
    ./ci/minikube_e2e_tests_inner.sh
```

File: ci/minikube_e2e_tests.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+TEST_RESULT=$?
+
+docker cp $CONTAINER_NAME:/container_workspace/artifacts/ $PWD
```
Comment (Victoremepunto) [inline ci/minikube_e2e_tests.sh:38]:
```suggestion
docker cp "${CONTAINER_NAME}:/container_workspace/artifacts/" .
```

File: ci/build_push_base_img.sh (hunk 4 above)
Comment (Victoremepunto) [inline ci/build_push_base_img.sh:21]:
```suggestion
    BASE_IMG="$BASE_IMG" make docker-build-and-push-base
```

**Review by Victoremepunto — APPROVED** [review]
I think it looks good, great job!

**Review by adamrdrew — APPROVED** [review]
Let's Get This Merged!

**Review by bsquizz — APPROVED** [review]
LGTM!

**Review by adamrdrew — APPROVED** [review]

**Review by Victoremepunto — APPROVED** [review]
LGTM

**Pull request merged**